kernel/exec.c

```diff
@@ -205,9 +224,9 @@ exec(const char *path, const char* const *argv)
 
   int result;
   if (readelfhdr(ip, &elf)) {
-    result = execelf(path, path, argv, &elf, &ip, &pgdir);
+    result = execelf(path, path, argv, envp, &elf, &ip, &pgdir);
   } else {
-    result = execshebang(exepath, argv, &ip, &pgdir);
+    result = execshebang(exepath, argv, envp, &ip, &pgdir);
```

Comment:
Would you replace `exepath` to `path`
(and remove `exepath` local variable)
when you removed `find_path`?